dragonfly/implementations/roach2interface.py

```diff
@@ -0,0 +1,316 @@
+            self.fft_shift_vector = (self.fft_shift, 'cd')
+        return self.configured
+
+
+    def get_calibration_status(self):
+        return self.calibrated
+
+
+    def get_configuration_status(self):
+        return self.configured
+
+
+    def do_adc_ogp_calibration(self, **kwargs):
+        logger.info('Calibrating ROACH2, this will take a while.')
+        logger.info('Doing adc ogp calibration')
+        adc_dictionary = ArtooDaq.calibrate_adc_ogp(self, **kwargs)
+        logger.info(adc_dictionary)
+        self.calibrated=True
+        return self.calibrated
+
+
+    @property
+    def is_running(self):
+        logger.info('Pinging ROACH2')
+        response = os.system("ping -c 1 " + self.roach2_hostname)
+        #and then check the response...
+        if response == 0:
+            logger.info('ROACH2 is switched on')
+
+            if not self.get_configuration_status:
+                logger.info('ROACH2 is not configured')
+            if not self.get_calibration_status:
+                logger.info('Roach2 is not calibrated')
+        else:
+            self.configured=False
+            self.calibrated=False
+
+        return self.calibrated
+
+
+    def block_channel(self, channel):
+        self.block_dict[channel]=True
+
+
+    def unblock_channel(self, channel):
+        self.block_dict[channel]=False
+
+
+    @property
+    def blocked_channels(self):
+        bc = [i for i in self.block_dict.keys() if self.block_dict[i]==True]
+        return bc
+
+
+    def get_central_frequency(self, channel):
+        return self.freq_dict[channel]
+
+
+    def set_central_frequency(self, cf, channel):
+        if self.block_dict[channel]==False:
+            logger.info('setting central frequency of channel {} to {}'.format(channel, cf))
+            cf = ArtooDaq.tune_ddc_1st_to_freq(self, cf, tag=channel)
+            self.freq_dict[channel]=cf
+            return cf
+        else:
+            raise core.exceptions.DriplineGenericDAQError('Channel {} is blocked'.format(channel))
+
+
+    @property
+    def all_central_frequencies(self):
+        return self.freq_dict
+            
+
+    @property
+    def gain(self):
+        return self.gain_dict
+
+
+    @gain.setter
+    def gain(self, val):
+        gain, channel= val
+        if self.block_dict[channel]==False:
+            if gain>-8 and gain <7.93:
+                logger.info('setting gain of channel {} to {}'.format(channel, gain))
+                ArtooDaq.set_gain(self, gain, tag=channel)
+                self.gain_dict[channel] = gain
+            else:
+                raise core.exceptions.DriplineGenericDAQError('Only gains between -8 and 7.93 are allowed')
+        else:
+            raise core.exceptions.DriplineGenericDAQError('Channel {} is blocked'.format(channel))
+
+
+    @property
+    def fft_shift_vector(self):
+        return self.fft_shift
+
+
+    @fft_shift_vector.setter
+    def fft_shift_vector(self, val):
+        shift, tag = val
```

Comment:
This is consistent with above, please add a doc string. As above I like the setter but wish it was consistent with the previous earlier source file.